is_equal/is_equal.tex

```diff
@@ -0,0 +1,155 @@
+    \item Function: returns vector mapping of whether vector elements equal input \texttt{value}.
+    \item Input Metric: Symmetric Distance
+    \item Output Metric: same as Input Metric. 
+    \item Stability Relation: function that takes in 32 bit integers $d_{in}$ and $d_{out}$ and returns whether $d_{in} \leq d_{out}.$
+\end{itemize}
+
+\silvia{Do we need to include the stability parameter? And perhaps include an example calling \texttt{make\_is\_equal} with metrics?}
+
+\silvia{Shouldn't it be checked / imposed that the end user passes the same input and out metric somewhere? Otherwise have only one?}
+
+\subsection{Proof}
+
+\begin{theorem}
+For every setting of the input parameters to \texttt{make\_is\_equal}, the transformation returned by \texttt{make\_is\_equal} has the following properties
+\begin{enumerate}
+    \item \textup{(Appropriate output domain).} If vector \texttt{v} of type \texttt{T} is in the input domain, then \texttt{function(v)} is in the output domain, otherwise the program will raise an \texttt{Exception}. 
+    \item \textup{(Stability Guarantee).} If two vector inputs $v, w$ are "$d_{in}$ - close" and if \texttt{Relation($d_{in}$, $d_{out}$) = True} then the corresponding outputs $function(v), function(w)$ are "$d_{out}$ - close". 
+\end{enumerate}
+\end{theorem}
+\begin{proof}
+\begin{enumerate}
+\item \textup{(Appropriate output domain).} Rust automatically enforces the type system to ensure it is in the correct output domain. The type signature of the function in the pseudo code and Rust code takes in a \texttt{Vec<T>} and returns \texttt{Vec<Bool>}. Since the rust code successfully compiles, by definition of the type signature the theorem must hold. Otherwise, the code will raise an exception for incorrect input type.
+
+
+
+\item \textup{(Stability Guarantee).}
+We can assume $d_{in} \leq d_{out}$ because Relation($d_{in}$, $d_{out}$) = \texttt{True}. We consider the symmetric distance as input metric. Since vector inputs $v, w$ are $d_{in}$-close, then by definition the symmetric distance of the multisets is bounded by $d_{in}$: $\abs{MultiSet(v) \Delta MultiSet(w)} \leq d_{in}$ 
+
+% We can denote the histogram of the multiset $v \in MultiSets(\mathcal{X})$ as $h_v : \mathcal{X} \to \mathbb{N}$ where $h_v(z)$ is the number of occurrences of $z$ in $\mathcal{X}$. $$d_{Sym}(v, w) = \norm{h_v - h_w}_1 = \sum_{z} \abs{h_v(z) - h_w(z)}.$$
+
+We want to show that $\abs{MultiSet(function(v)) \Delta MultiSet(function(w))} \leq d_{out}$. %\\$\abs{MultiSet(function(v))\Delta MultiSet(function(w))} = \norm{h_{is\_equal(v)} - h_{is\_equal(w)}}_1 \leq d_{out}$. 
+To do so, we consider the symmetric distance between the multiset transformations $function(v)$ and $function(w)$. We claim that for any pair of set elements $v^* \in Multiset(v)$ and $w^* \in Multiset(w)$ $$\abs{MultiSet(function(v)) \Delta MultiSet(function(w))} \leq \abs{MultiSet(v) \Delta MultiSet(w)}$$
+We consider 4 cases. 
```

Comment:
this sentence doesn't parse.  It says "We claim that for any pair v^* and w^*, STATEMENT holds", but STATEMENT makes no reference to v^* and w^*.

Reply:
I think you are trying to use the Path Property for SymmetricDistance, but you need to do so formally (referring to the defs and lemmas in the proof_defns doc) if you want to do that.  Else you can follow the simplified proof outline I suggested in clamp.tex.

Reply:
Got it, will look at path property to formalize this proof.

Reply:
You may find the proof strategy suggested in clamp.tex to be simpler than using the path property.